scipy/linalg/tests/test_basic.py

```diff
@@ -1334,6 +1334,46 @@ def test_axis_args(self):
         assert_equal(x.shape, (4, 2, 3))
         assert_allclose(x, expected)
 
+class TestBalance(TestCase):
+    
+    def test_string_arg(self):
+        assert_raises(ValueError, balance, 'Some string for fail')
+        
+    def test_infnan_arg(self):
+        assert_raises(ValueError, balance, np.array([[1,2],[3,np.inf]]))
+        assert_raises(ValueError, balance, np.array([[1,2],[3,np.nan]]))        
+        
+    def test_scaling(self):
+        _, y = balance(np.array([[1000, 1], [1000, 0]]))
+        assert_allclose(int(np.diff(np.log2(np.diag(y)))), 5)
+    
+    def test_scaling_order(self):
+        A = np.array([[1, 0, 1e-4],[1, 1, 1e-2],[1e4, 1e2, 1]])
+        x, y = balance(A)
+        assert_allclose(solve(y,A).dot(y),x)
+
+    def test_separate(self):
+        _, (y, z) = balance(np.array([[1000, 1], [1000, 0]]), separate=1)
+        assert_equal(int(np.diff(np.log2(y))), 5)
+        assert_allclose(z,np.arange(2))
+        
+    def test_permutation(self):
+        A = block_diag(np.ones((2,2)),np.tril(np.ones((2,2))),np.ones((3,3)))
```

Comment:
same comment for 2 and 3 lines lower, and the assert in the next test case below